src/coreclr/vm/prestub.cpp

```diff
@@ -1067,7 +1054,436 @@ PCODE MethodDesc::JitCompileCodeLocked(PrepareCodeConfig* pConfig, JitListLockEn
+        UINT targetArgCount = stubArgCount - beginIndex;
+        UINT targetRetCount = cxt.DeclarationSig.IsReturnTypeVoid() ? 0 : 1;
+        switch (cxt.Kind)
+        {
+        case UnsafeAccessorKind::Constructor:
+            _ASSERTE(cxt.TargetMethod != NULL);
+            pCode->EmitNEWOBJ(pCode->GetToken(cxt.TargetMethod), targetArgCount);
+            break;
+        case UnsafeAccessorKind::Method:
+            _ASSERTE(cxt.TargetMethod != NULL);
+            pCode->EmitCALLVIRT(pCode->GetToken(cxt.TargetMethod), targetArgCount, targetRetCount);
+            break;
+        case UnsafeAccessorKind::StaticMethod:
+            _ASSERTE(cxt.TargetMethod != NULL);
+            pCode->EmitCALL(pCode->GetToken(cxt.TargetMethod), targetArgCount, targetRetCount);
+            break;
+        case UnsafeAccessorKind::Field:
+            _ASSERTE(cxt.TargetField != NULL);
+            pCode->EmitLDFLDA(pCode->GetToken(cxt.TargetField));
+            break;
+        case UnsafeAccessorKind::StaticField:
+            _ASSERTE(cxt.TargetField != NULL);
+            pCode->EmitLDSFLDA(pCode->GetToken(cxt.TargetField));
+            break;
+        default:
+            _ASSERTE(!"Unknown UnsafeAccessorKind");
+            return false;
+        }
+
+        // Return from the generated stub
+        pCode->EmitRET();
+
+        {
+            UINT maxStack;
+            size_t cbCode = sl.Link(&maxStack);
+            DWORD cbSig = sl.GetLocalSigSize();
+
+            COR_ILMETHOD_DECODER* pILHeader = ilResolver->AllocGeneratedIL(cbCode, cbSig, maxStack);
+            BYTE* pbBuffer = (BYTE*)pILHeader->Code;
+            BYTE* pbLocalSig = (BYTE*)pILHeader->LocalVarSig;
+            _ASSERTE(cbSig == pILHeader->cbLocalVarSig);
+            sl.GenerateCode(pbBuffer, cbCode);
+            sl.GetLocalSig(pbLocalSig, cbSig);
+
+            // Store the token lookup map
+            ilResolver->SetTokenLookupMap(sl.GetTokenLookupMap());
+
+            *resolver = (DynamicResolver*)ilResolver;
+            *methodILDecoder = pILHeader;
+        }
+
+        ilResolver.SuppressRelease();
+        return true;
+    }
+}
+
+bool MethodDesc::TryGenerateUnsafeAccessor(DynamicResolver** resolver, COR_ILMETHOD_DECODER** methodILDecoder)
+{
+    STANDARD_VM_CONTRACT;
+    _ASSERTE(resolver != NULL);
+    _ASSERTE(methodILDecoder != NULL);
+    _ASSERTE(*resolver == NULL && *methodILDecoder == NULL);
+    _ASSERTE(IsIL());
+    _ASSERTE(GetRVA() == 0);
+
+    // The UnsafeAccessorAttribute is applied to methods with an
+    // RVA of 0 (for example, C#'s extern keyword).
+    const void* data;
+    ULONG dataLen;
+    HRESULT hr = GetCustomAttribute(WellKnownAttribute::UnsafeAccessorAttribute, &data, &dataLen);
+    if (hr != S_OK)
+        return false;
+
+    UnsafeAccessorKind kind;
+    SString name;
+
+    CustomAttributeParser ca(data, dataLen);
+    if (!TryParseUnsafeAccessorAttribute(this, ca, kind, name))
+        ThrowHR(COR_E_BADIMAGEFORMAT);
```

Comment:
When people see this exception, [BadImageFormatException docs](https://learn.microsoft.com/en-us/dotnet/api/system.badimageformatexception) will send them chasing all wrong problems. It would be nice to show a message that gives a hint that the problem is related to the UnsafeAccessorAttribute. It does not need to be super detailed. 

Something like "Invalid usage of UnsafeAccessorAttribute."